src/main/java/net/helpscout/api/ApiClient.java

```diff
@@ -1442,52 +1429,23 @@ private void setThreadProperties(ConversationThread thread) {
 
     private <T> Page<T> getPage(String url, Map<String,String> params, Class<T> clazzType, int expectedCode) throws ApiException {
         url = setParams(url, params);
-        String json = doGet(url, HTTP_STATUS_OK);
+        String json = httpMethodWrapper.doGet(url, HTTP_STATUS_OK);
         JsonElement obj = parseJson(url, json);
         
         return JSONUtils.objectToPage(obj.getAsJsonObject(), clazzType);
     }
-    
-    private <T> T doPost(String url, String requestBody, int expectedCode, ResultExtractor<T> extractor) throws ApiException {
-
-        try (HTTPConnectionWrapper conn = new HTTPConnectionWrapper(apiKey, baseUrl + url, METHOD_POST, expectedCode, requestBody)) {
-            return extractor.extract(conn);
-        } catch(ApiException ex) {
-            throw ex;
-        } catch (Exception ex) {
-            throw new RuntimeException(ex);
-        }
-    }
-
-    private void doPut(String url, String requestBody, int expectedCode) throws ApiException {
 
-        try (HTTPConnectionWrapper conn = new HTTPConnectionWrapper(apiKey, baseUrl + url, METHOD_PUT, expectedCode, requestBody)) {
-        } catch(ApiException ex) {
-            throw ex;
-        } catch (Exception ex) {
-            throw new RuntimeException(ex);
-        }
-    }
+    private <T> ArrayList<T> getPageItems(String url, Class<T> clazzType) throws ApiException {
```

Comment:
I think it would be better if the return type was `List<T>` and not `ArrayList<T>`, also `clazzType` could be named after what it really represents, for example `responseType`